src/client/client/intercept_readlink.c

```diff
@@ -31,19 +31,25 @@ Place, Suite 330, Boston, MA 02111-1307 USA
 ssize_t (*orig_readlink)(const char *path, char *buf, size_t bufsiz);
 ssize_t (*orig_readlinkat)(int dirfd, const char *pathname, char *buf, size_t bufsiz);
 
-extern char *location;
+static char *cachepath;
+
+void set_intercept_readlink_cachepath( char *chosen_realized_cachepath, char *chosen_parsed_cachepath, char *chosen_symbolic_cachepath ){
+    cachepath = chosen_realized_cachepath;
+    chosen_parsed_cachepath = chosen_parsed_cachepath;
+    chosen_symbolic_cachepath = chosen_symbolic_cachepath;
+}
```

Comment:
I'm a bit confused by the set_intercept_readlink_cachepath() and set_should_intercept_cachepath(). Seems like we're keeping multiple copies of the paths in different static variables, then calling each of these functions to set the different copies.

Different variables holding the same copies of information are a source of bugs. Let's consolidate the variables holding paths in the clients to just one instance.

Or am I missing something where these are different?

Reply:
Two issues here:  
1) We're using setter functions to pass around pointers to static file-scoped variables.  Making the original variable global gets rid of the copies, but creates its own source of bugs.

2) We're keeping track of the realized, parsed, and symbolic versions of the cachepaths, Parsed and symbolic aren't used in `set_inercept_readlink_cachepath()` and symbolic isn't used in `set_should_intercept_cachepath()`.  (The code that references them in the setter function is there to silence the compiler warning.)

Let's err on the side of deleting more code.  Removed the two `set_*()` functions.